### DISCUSSION TITLE
no hashes in filenames when using js-file as entrypoint

### DISCUSSION THREAD

**Discussion by scsskid** [parcel-bundler/parcel#7559] [Q&A]
when using a js file instead of html file as entrypoint, no hashes are appended or manifest.json is generated

what setup is recommended, when using parcel to implement in a simple php project, where i want only to generate 

styles.[hash].css
scripts.[hash].css
manifest.json

thank you!

**Comment by andre-krueger** [comment]
You can use that plugin: https://github.com/ol-loginov/parcel-namer-rewrite
I'm using this config in package.json (latest Parcel version, 2.5.0) to get the hashes in the filenames (feels wrong, though 😄 ):

```
  "parcel-namer-rewrite": {
    "chain": "@parcel/namer-default",
    "rules": {
      "(.*)\\.(js|css)": "$1.{hash}.$2"
    }
  }
```
Unfortunately, this doesn't work in combination with [this plugin](https://github.com/autifyhq/parcel-reporter-bundle-manifest) (the JS/CSS filenames don't have hashes in the values of the manifest), so I have to resort to use HTML as entrypoints again.

@devongovett 
Is there a reason why non-HTML entry files don't get hashes?

It would be super helpful to be able to use an asset manifest to serve the correct assets in a server side rendered app (Django templates via [static file helper](https://docs.djangoproject.com/en/4.0/howto/static-files/), for example).

This static helper approach isn't working, of course, as Parcel wouldn't be able to resolve something like this: `<link href="{{static "index.css"}}" rel="stylesheet" />`.

Currently I have to use a workaround by using HTML templates as an entry point for Parcel and then copy these transformed templates (with hashed asset names) to the backend template directory.

**Reply by mischnic** [reply]
> Is there a reason why non-HTML entry files don't get hashes?

Entrypoints don't get hashed because otherwise you wouldn't be able to differentiate the entrypoint in the distdir from other assets/shared bundles. (This is also the reason why HTML entry points don't get a hash)

For hashed entrypoints, Parcel would indeed have to output an additional manifest that tells you which files are the entrypoints. That's something we want to facilitate at some point: https://github.com/parcel-bundler/parcel/issues/4227.
